### ISSUE TITLE
Authentication asked to access artifacts folder (without any authentication set up)

### ISSUE THREAD

**Issue by Diogomrol** [gocd/gocd#4774]
Hello there,

Before (at least in GoCD 16.2.1) users could access artifact folders generated by the pipelines via URL, but now they are prompted to authenticate. We have no authentication configured, so this was not expected.

This only happens when trying to access a folder and not a specific file.

Example: 

This one works (accessing a specific file):
http://<go_url_here>/go/files/Build_Trunk/1/Build/1/Build/build_reports/build_output_PlatformServer.txt

This one asks for authentication (accessing a folder):
http://<go_url_here>/go/files/Build_Trunk/1/Build/1/Build/build_reports

This is happening in the latest version (GoCD 18.5.0).

**Comment by varshavaradarajan** [comment]
@Diogomrol - can you paste the exact request and the response from the gocd server when you make the request for a directory? 

Also note that you need to request for directory.zip as the gocd server will compress the directly for downloading.

**Comment by Diogomrol** [comment]
Hello,

I tried this request using the browser. It used to work before (and without the .zip), that's why it looks like a behavior change (unless this was never intended to work in the first place).

When I try to access that URL I first see this (as I used to):

![imagem](https://user-images.githubusercontent.com/9485620/40437474-59ae53b8-5ead-11e8-8e6a-4d601cf1b3e2.png)

And then I immediatelly after get redirected to authenticate.

**Comment by arvindsv** [comment]
As far as I know, it was always supposed to be `.zip`.

https://api.gocd.org/current/#get-artifact-directory
https://api.gocd.org/17.10.0/#get-artifact-directory

**Comment by Diogomrol** [comment]
Right :)

Probably we were taking advantage of something "undocumented" by trying to access the folder directly.

Anyway, was it supposed to ask for authentication when we have no authentication at all?

**Comment by varshavaradarajan** [comment]
> And then I immediatelly after get redirected to authenticate.

I think we setup a redirection to the login page for urls if unauthenticated. But its not supposed to do that for a gocd server where security is not configured. 

When I tried the request for directory with a fresh installation of 18.5.0, with no security configured, I got the html page that you pasted above. I need to check the request on the browser to see the redirection happening. 

We have recently upgraded spring security for 18.6.0. May be its worthwhile verifying this behaviour on an experimental installer as well. cc @bdpiparva , @ketan
